### ISSUE TITLE
InceptionResnetV1 batch normalization layer

### ISSUE THREAD

**Issue by 15m43lk4155y** [timesler/facenet-pytorch#116]
I don't know if there is a way to feed non-batched data to the model without modifying the source code but I think it should be handled because the last batch normalization layer raises an error if the batch size is equal to 1. if it were to be modified to conditionally on the batch size is 1 the BatchNorm1d layer would be replaced by InstanceNorm1d or else it stays the same, that would be better.

**Comment by timesler** [comment]
@15m43lk4155y are you training or fine-tuning the model or just using it for inference/testing? If the latter, batch norm will work fine for a batch size of 1 provided the model is in `eval()` mode.

We could quite easily add an argument `norm_layer` for passing a custom normalisation layer in a similar way to torchvision models. If that is the functionality you would like, it would be great if you could make a PR with the change.

**Comment by ctivanovich** [comment]
@15m43lk4155y You could also drop the last batch in your loader; there is an optional loader argument to do that.
